app/controllers/course/settings_controller.rb

```diff
@@ -13,8 +14,26 @@ def update #:nodoc:
     end
   end
 
+  def components #:nodoc:
+  end
+
+  def update_components #:nodoc:
+    @settings.update(params.require(:settings_effective))
```

Comment:
The param name is same as instance yah ?
https://github.com/Coursemology/coursemology2/blob/master/app/controllers/admin/admin_controller.rb#L12